apps/frontend/src/libs/components/date-input/date-input.tsx

```diff
@@ -0,0 +1,109 @@
+import { Calendar } from "react-calendar";
+import {
+	type Control,
+	type FieldPath,
+	type FieldValues,
+} from "react-hook-form";
+
+import { Icon, Popover } from "~/libs/components/components.js";
+import { formatDate } from "~/libs/helpers/helpers.js";
+import {
+	useCallback,
+	useFormController,
+	usePopover,
+} from "~/libs/hooks/hooks.js";
+
+import { addDays, getDifferenceInDays } from "./libs/helpers/helpers.js";
+import { type DateValue } from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+type Properties<T extends FieldValues> = {
+	control: Control<T, null>;
+	maxRange?: number;
+	name: FieldPath<T>;
+};
+
+const DateInput = <T extends FieldValues>({
+	control,
+	maxRange,
+	name,
+}: Properties<T>): JSX.Element => {
+	const { field } = useFormController({
+		control,
+		name,
+	});
+
+	const { isOpened, onClose, onOpen } = usePopover();
+
+	const getSelectedDateRange = (): string => {
+		const [startDate, endDate] = field.value as Date[];
+
+		return `${formatDate(startDate as Date, "MMM d, yyyy")} - ${formatDate(endDate as Date, "MMM d, yyyy")}`;
+	};
+
+	const handleDateChange = useCallback(
+		(dates: DateValue): void => {
+			const [startDate, endDate] = dates as Date[];
+
+			if (maxRange && startDate && endDate) {
+				const diffInDays = getDifferenceInDays(endDate, startDate);
+
+				if (diffInDays > maxRange) {
+					const adjustedEndDate = addDays(startDate, maxRange);
+					field.onChange([startDate, adjustedEndDate]);
+
+					return;
+				}
+			}
+
+			field.onChange(dates);
+		},
+		[field, maxRange],
+	);
+
+	const formatShortWeekday = useCallback(
+		(_locale: string | undefined, date: Date): string => {
+			return formatDate(date, "EEEEEE");
```

Comment:
Maybe we can store date formats somewhere in enums, as we do with DataStatus for instance?

Reply:
You're right, I'll try to change it, thanks

Reply:
Okay, now I'm not so sure about it, the naming for these formats is quite complicated, because it either will need a lot of words, or if we will use something like "short date"/"long date" it won't be enough, because we might have a lot of similar formats. I guess that this type is enough for now.